### PR TITLE
 French layout improvement suggestion

### PR DESCRIPTION
**Pull request by breversa**
Following https://github.com/dessalines/thumb-key/issues/125

### CODE REVIEW

**Comment by dessalines** [comment]
@CastelJeremy You already really did a review, but after @breversa updates the PR for your requested changes, hit approve and I'll know when this is ready to merge.

**Comment by breversa** [comment]
@dessalines I've added a new commit with some needed changes.

What can I do now to have this layout implemented into Thumb-Key?

**Comment by breversa** [comment]
@dessalines:
Since @CastelJeremy seems not to be around at the moment, can anyone else approve this?

**Review by dessalines — CHANGES_REQUESTED** [review]
You still have a lot of symbols on this keyboard, which should not be there, and added to the numeric keyboard, as part of a separate PR.

**Comment by breversa** [comment]
I know ; I'd like to test that layout as it is before trimming it down.
If it helps, I'm fine with it being implemented as an "alternative" french layout rather that a "v2" french layout. :)

**Comment by dessalines** [comment]
Please make the changes requested. Do not put symbols on anything but the numeric keyboard.

**Comment by CastelJeremy** [comment]
Sorry for the late reply.

I agree with @dessalines, we should focus on the letter layout.

Also the pipeline is currently failing because of formatting errors, could you update your code to fix them ?
https://woodpecker.join-lemmy.org/dessalines/thumb-key/pipeline/90/3

Finally, we should stick to naming this update the V2 and as such you might need to update the following lines :

https://github.com/dessalines/thumb-key/blob/ecad1f328eb93a6a082cb0554bcd60034a84e8df/app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt#L691
https://github.com/dessalines/thumb-key/blob/56a72984f03941f2e4b5666b1674d3c7f101c7c6/app/src/main/java/com/dessalines/thumbkey/utils/Types.kt#L95
https://github.com/dessalines/thumb-key/blob/56a72984f03941f2e4b5666b1674d3c7f101c7c6/app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt#L43
https://github.com/dessalines/thumb-key/blob/56a72984f03941f2e4b5666b1674d3c7f101c7c6/app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt#L86

File: app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt [hunk 1]
```diff
@@ -174,58 +212,33 @@ val THUMBKEY_FR_V1_MAIN = KeyboardC(
...
                 ),
                 swipes = mapOf(
                     SwipeDirection.TOP_RIGHT to KeyC(
-                        display = KeyDisplay.TextDisplay("p"),
-                        action = KeyAction.CommitText("p")
-                    )
-                )
+                        display = KeyDisplay.TextDisplay("m"),
+                        action = KeyAction.CommitText("m")
+                    ),
```
Comment (CastelJeremy) [inline app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt:224]: Missing a closing parenthesis right after this line.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt [hunk 2]
```diff
@@ -174,58 +212,33 @@ val THUMBKEY_FR_V1_MAIN = KeyboardC(
...
+                        action = KeyAction.CommitText("ç"),
                         color = ColorVariant.MUTED
                     ),
-                    SwipeDirection.LEFT to KeyC(
-                        display = KeyDisplay.TextDisplay(","),
-                        action = KeyAction.CommitText(","),
-                        color = ColorVariant.MUTED
+                    SwipeDirection.TOP to KeyC(
+                        display = KeyDisplay.TextDisplay("c"),
+                        action = KeyAction.CommitText("c")
```
Comment (CastelJeremy) [inline app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt:241]: You can leave the `ç` on the TOP LEFT but you don't have to remove the default symbols which are standard on other layouts (`'`, `,`, `*`, `.`, `-`)

File: app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt [hunk 3]
```diff
@@ -441,58 +495,34 @@ val THUMBKEY_FR_V1_SHIFTED = KeyboardC(
...
-                        color = ColorVariant.MUTED
-                    ),
-                    SwipeDirection.BOTTOM_LEFT to KeyC(
-                        display = KeyDisplay.TextDisplay("*"),
-                        action = KeyAction.CommitText("*"),
+                    SwipeDirection.TOP_LEFT to KeyC(
+                        display = KeyDisplay.TextDisplay("Ç"),
+                        action = KeyAction.CommitText("Ç"),
                         color = ColorVariant.MUTED
                     ),
```
Comment (CastelJeremy) [inline app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt:522]: Update this section with the changes done on the `MAIN` layout. (Restoring default symbols)

**Review by CastelJeremy — CHANGES_REQUESTED** [review]
Hey,
I reviewed your changes and added some comments where needed.

If you have some time, could you also look into my previous comment and update the `Utils.kt` and `Types.kt` files ? https://github.com/dessalines/thumb-key/pull/126#issuecomment-1509924679

Thank you for your work, now i really want to test this layout :D.

**Comment by dessalines** [comment]
This has gone a while without updates, If someone wants to make a new PR, feel free.

**Comment by breversa** [comment]
Sorry, I've been very busy lately, and didn't expect that a quick improvement suggestion would turn into a "how does Git work again?" marathon.

**Comment by breversa** [comment]
Sorry, I've been very busy lately, and didn't expect that a quick improvement suggestion would turn into a "how does Git work again?" marathon.

**Comment by dessalines** [comment]
Sure, feel free to open a new PR.

**Comment by breversa** [comment]
Is this correct now: app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyFRv2.kt ?

**Comment by dessalines** [comment]
I'll get to your PR within a few days.

**Pull request closed without merging**